crates/bevy_input/src/mouse.rs

```diff
@@ -65,3 +86,47 @@ pub fn mouse_button_input_system(
         }
     }
 }
+
+/// Updates the Axis<Mouse::Horizontal> and Axis<Mouse::Vertical> with raw data
+/// Can be different on other platforms
+/// Should be used for in-game movement with mouse
+pub fn mouse_axis_system(
+    mut state: Local<MouseMovementState>,
+    mut mouse_axis: ResMut<Axis<Mouse>>,
+    mouse_movement_events: Res<Events<MouseMotion>>,
+) {
+    for event in state
+        .mouse_movement_event_reader
+        .iter(&mouse_movement_events)
+    {
+        mouse_axis.add(Mouse::Horizontal, event.delta.x());
```

Comment:
Using `add()` in these systems results in broken output. In the example, i get gigantic and/or incorrect numbers:
```rs
mouse: [x: :-1843, y: :321] cursor: [x: :4682519.5, y: :2292061.5]
```

I'm pretty sure we want `set()`

Reply:
no, `add` adds mouse delta. I've tried this in my simple FPS example and this is correct. You need to multiply by sensitivity which is about 0.05 to get smooth mouse movement, but this works for proper rotation. Set wouldn't work for mouse_delta

Reply:
First, its right to call out that "mouse delta" is a little different from the other inputs. I think, if anything, we should consider removing mouse delta from the axis api as it doesn't really fit into the same "bucket" as the other inputs. We might have a number of "mouse delta" events per frame, so we can't consider them individually to determine the "state" of the mouse. I would argue that there is no "fixed state" of the mouse. Only a list of "relative changes in state" each frame.

In my mind the purpose of the Axis api is to provide a simple interface to consume raw inputs that represent the **current** state of an axis. For example, if we add joysticks to this api, I would want the x-axis to be in the range [-1, 1].  Users could then use those numbers to determine how far the joystick has been tilted to the left or right (and adjust things like character speed based on that state). Currently all of the systems use `add()`, including "non delta" systems. This results in very large, "aggregate" numbers. In the joystick example, I wouldn't want it to increment to very large numbers as I wiggle the joystick, as that prevents the data from being used to determine "how fast should i move this character this frame and in what direction?".

The current aggregate mouse delta works for rotation in your fps because rotations can "wrap around". If you want that behavior, I think you should maintain an aggregation in a separate resource. But it would probably be better to take the raw "delta" events, multiply them by some weight, and rotate the camera directly (relative to its current position). That way you don't need to maintain an aggregation (which is pretty meaningless across frames).

Reply:
But what about Cursor? Cursor also returns delta and not it's actual position. That is the same with joysticks. Should they use aggregate function? Because otherwise it'll be something like `0.0 0.0 33.0 0.1 0.1 -34.0 0.1...` I don't see why this should not be aggregate.

Reply:
Gilrs joystick events are absolute values in the range [-1, 1]: https://docs.rs/gilrs/0.7.4/gilrs/ev/enum.EventType.html#variant.AxisChanged. For example, -1 is "all the way left" and 1 is "all the way right"

Winit's CursorMoved event is also an absolute value: https://docs.rs/winit/0.22.2/winit/event/enum.WindowEvent.html#variant.CursorMoved

Reply:
okay, have read that wrong. Then I have no objections, it didn't make sense.